consensus/src/lib.rs

```diff
@@ -505,20 +514,21 @@ pub struct Proposer<C: Send + Sync, TxApi: PoolChainApi> where
 	parent_number: BlockNumber,
 	tracker: Arc<AttestationTracker>,
 	transaction_pool: Arc<Pool<TxApi>>,
+	slot_duration: SlotDuration,
 }
 
-impl<C, TxApi> consensus::Proposer<Block, AuraConsensusData> for Proposer<C, TxApi> where
+impl<C, TxApi> consensus::Proposer<Block> for Proposer<C, TxApi> where
 	TxApi: PoolChainApi<Block=Block>,
 	C: ProvideRuntimeApi + HeaderBackend<Block> + Send + Sync,
-	C::Api: ParachainHost<Block> + BlockBuilderApi<Block, InherentData>,
+	C::Api: ParachainHost<Block> + BlockBuilderApi<Block>,
 {
 	type Error = Error;
 	type Create = Either<
 		CreateProposal<C, TxApi>,
 		future::FutureResult<Block, Error>,
 	>;
 
-	fn propose(&self, consensus_data: AuraConsensusData) -> Self::Create {
+	fn propose(&self, inherent_data: InherentData, max_duration: Duration) -> Self::Create {
```

Comment:
@bkchr can you review this function to make sure the inherent data is backwards compatible with before?

Reply:
Do we need to produce blocks with the new node and the old runtime? If yes, then this is not compatible. For syncing, this code should not be touched.

Reply:
Ok, thanks for the clarification. I think we can work around this. What would happen if the old code tried to author on the new runtime?

Reply:
Create inherent will fail and thus the block production will fail.